qos-service-impl/src/main/java/com/palantir/atlasdb/qos/ratelimit/RateLimiter.java

```diff
@@ -0,0 +1,306 @@
+/*
```

Comment:
Copied from Guava

Reply:
doesn't this need to keep Apache license?

Reply:
whoops, good catch. do you know if keeping the license header ad the top of the file is sufficient?